core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala

```diff
@@ -117,6 +119,28 @@ private[ui] class ExecutorsPage(
     val maximumMemory = info.maxMemory
     val memoryUsed = info.memoryUsed
     val diskUsed = info.diskUsed
+
+    // Determine Color Opacity from 0.5-1
+    var activeTasksAlpha = 1.0
+    var failedTasksAlpha = 1.0
+    var completedTasksAlpha = 1.0
+    var totalDurationAlpha = 1.0
+    if (info.totalCores > 0) {
+      // activeTasks range from 0 to all cores
+      activeTasksAlpha = (info.activeTasks.toDouble / info.totalCores) * 0.5 + 0.5
+    }
+    if (info.totalTasks > 0) {
```

Comment:
You can also assign these like
```
val (failedTasksAlpha, completedTasksAlpha) = 
  if (...) {
    (..., ...)
  } else {
    (1.0, 1.0)
  }
```
but whether that's clearer is a matter of taste.
It might be better to cap alpha at 1.0 just for tidiness.